### PR TITLE
feat(core): Policies

### PR DESCRIPTION
**Pull request by robzienert**
cc @emjburns @robfletcher @cfieber @jeyrschabu

### CODE REVIEW

**Comment by robzienert** [comment]
Should be noted, this doesn't wire up policies for use anywhere. That will be done in a forthcoming PR from Emily.

File: keel-web/src/main/kotlin/com/netflix/spinnaker/keel/controllers/v1/admin/PolicyController.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import com.netflix.spinnaker.keel.Policy
+import com.netflix.spinnaker.keel.PolicyRepository
+import com.netflix.spinnaker.keel.matcher.AllMatcher
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+@RequestMapping("/v1/admin")
```
Comment (emjburns) [inline keel-web/src/main/kotlin/com/netflix/spinnaker/keel/controllers/v1/admin/PolicyController.kt]: What do you think about making this more descriptive? Like `/v1/policy`. Or, if you envision more admin things, changing the requests to be `/v1/admin/policy`

File: keel-web/src/main/kotlin/com/netflix/spinnaker/keel/controllers/v1/admin/PolicyController.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+class PolicyController
+@Autowired constructor(
+  private val policyRepository: PolicyRepository
+) {
+
+  fun list() = policyRepository.findAll()
+
```
Comment (emjburns) [inline keel-web/src/main/kotlin/com/netflix/spinnaker/keel/controllers/v1/admin/PolicyController.kt]: Don't you need `@RequestMapping(method = arrayOf(RequestMethod.POST))`? And similar ones for all these requests.

**Comment by emjburns** [comment]
LGTM

File: keel-web/src/main/kotlin/com/netflix/spinnaker/keel/controllers/v1/admin/PolicyController.kt (hunk 1 above)
Comment (robzienert) [reply keel-web/src/main/kotlin/com/netflix/spinnaker/keel/controllers/v1/admin/PolicyController.kt]: Derp, yep - that's what I meant to do `/v1/admin/policy`. Will fix.

File: keel-web/src/main/kotlin/com/netflix/spinnaker/keel/controllers/v1/admin/PolicyController.kt (hunk 2 above)
Comment (robzienert) [reply keel-web/src/main/kotlin/com/netflix/spinnaker/keel/controllers/v1/admin/PolicyController.kt]: Also derp.

**Pull request merged**